src/res_parsetree_viewer.ml

```diff
@@ -177,6 +184,17 @@ let funExpr expr =
       pexp_attributes = attrs;
     } as expr ->
     collect attrs [] {expr with pexp_attributes = []}
+  | {
+      pexp_desc = Pexp_fun (_, _defaultExpr, _pattern, _returnExpr);
+      pexp_attributes = attrs;
+    } as expr
+    when hasAsyncAttribute attrs ->
+    let asyncAttr =
+      List.filter
+        (fun (({txt}, _) : Parsetree.attribute) -> txt = "res.async")
+        attrs
+    in
+    collect asyncAttr [] {expr with pexp_attributes = []}
```

Comment:
This throws away all the other attributes, if there are some.

Reply:
You are correct.